addons/pos_razorpay/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -23,4 +24,36 @@ patch(PaymentScreen.prototype, {
             }
         });
     },
+    addNewPaymentLine(paymentMethod) {
+        if (paymentMethod.use_payment_terminal === "razorpay" && this.isRefundOrder) {
+            const refundedOrder = this.currentOrder.lines[0]?.refunded_orderline_id?.order_id;
+            const razorpayPaymentlines = refundedOrder.payment_ids.filter(
+                (pi) => pi.payment_method_id.use_payment_terminal === "razorpay"
+            );
+            const res = super.addNewPaymentLine(paymentMethod);
+            const newPaymentLine = this.paymentLines.at(-1);
+            if (
+                res &&
+                (razorpayPaymentlines.length !== 1 ||
+                    Math.abs(newPaymentLine.amount) < razorpayPaymentlines[0].amount ||
+                    newPaymentLine.amount === 0)
+            ) {
+                this.deletePaymentLine(newPaymentLine.uuid);
+                this.pos.notification.add(
+                    _t(
+                        "Adding a new Razorpay payment line is not allowed under the current conditions."
+                    ),
+                    { type: "warning", sticky: false }
+                );
+                return false;
+            }
+            if (res) {
+                newPaymentLine.setAmount(-razorpayPaymentlines[0].amount);
+                newPaymentLine.updateRefundPaymentLine(razorpayPaymentlines[0]);
+            }
```

Comment:
I don't understand what these lines try to accomplish. The newly added razor payment line already has the correct amount. Perhaps it should only do `updateRefundPaymentLine`.

If your intention is to prevent addition of new line if there is already a pending refund, then that should be the code:

```js
if (this.hasPendingRefundPaymentLine()) {
    // don't add a line by NOT calling super
    // and perhaps show a warning notification.
}
```